src/main.ts

```diff
@@ -508,20 +503,14 @@ ipcMain.handle(
 ipcMain.handle(
   'generator:open',
   async (_, fileName: string): Promise<GeneratorFileData> => {
-    let fileHandle: FileHandle | undefined
-
-    try {
-      fileHandle = await open(path.join(GENERATORS_PATH, fileName), 'r')
+    const data = await readFile(path.join(GENERATORS_PATH, fileName), {
```

Comment:
if we pass a file that doesn't exist it will fail in the schema parsing section right ?
Is the error correct in that case ? 🤔

Reply:
No, `readFile` should throw an error in this case.
> Is the error correct in that case ? 🤔

Do you mean the notification we show to the user? Currently, it says "Failed to load generator", which would still be more or less correct. It might be a good idea to have a distinction between fs errors/schema errors/anything else